src/DataTable/Table.tsx

```diff
@@ -66,7 +66,10 @@ const StyledTable = styled.table<React.ComponentPropsWithoutRef<'table'>>`
   .TableHeader,
   .TableCell {
     text-align: start;
+    display: flex;
+    align-items: center;
```

Comment:
I'm a little worried about how this might affect the readability of rows where some cells have 1 line of text, and other cells wrap to 2+ lines.

Compare these two tables:
<img width="1024" alt="vertically centered rows" src="https://github.com/primer/react/assets/2313998/a6087d58-b085-437b-b1f4-5ea86aa0f846">

<img width="1017" alt="flush top rows" src="https://github.com/primer/react/assets/2313998/8d380584-d603-4952-8afa-717907fd4684">

Reply:
Yeah that's a good point.  I'm not sure if the best solution is something that involves proper padding if there's a row action available, or increasing the base height of a row when there is no action.  I don't think anyone but us is using `DataTable` at the moment, and our current plan is to truncate row content when it overflows a cell.

Reply:
Late on this but I think `align-items: baseline` would solve for that specific issue, though not sure if it would introduce problems with other elements like IconButtons or Labels.